ang/civiawards/directives/award-stages-table.directive.js

```diff
@@ -0,0 +1,59 @@
+(function (angular, $, _) {
+  var module = angular.module('civiawards');
+
+  module.directive('civiawardAwardStagesTable', function () {
+    return {
+      controller: 'CiviawardAwardStagesTableController',
+      templateUrl: '~/civiawards/directives/award-stages-table.directive.html',
+      restrict: 'E'
+    };
+  });
+
+  module.controller('CiviawardAwardStagesTableController', function ($rootScope, $scope, CaseStatus) {
+    $scope.awardStages = CaseStatus.getAll();
+    $scope.toggleAwardStage = toggleAwardStage;
+
+    (function init () {
+      $rootScope.$on('civiawards::edit-award::details-fetched', setDetails);
+    }());
+
+    /**
+     * Toggle the selection of sent Award Stage
+     *
+     * @param {object} awardStage award stage object to be toggled
+     */
+    function toggleAwardStage (awardStage) {
+      $scope.basicDetails.selectedAwardStages[awardStage.value] =
+        !$scope.basicDetails.selectedAwardStages[awardStage.value];
+    }
+
+    /**
+     * Set Details
+     *
+     * @param {*} e event
+     * @param {object} details details of the award
+     */
+    function setDetails (e, details) {
```

Comment:
please use `event` instead of `e`. No single letter variable unless they are `x`, `y`, `z`, or `i` (and I'd even consider using no `i`).